lib/models/cli-command-completer.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+var fs          = require('fs');
+var find        = require('lodash/collection/find');
+var omelette    = require('omelette');
+var Generator   = require('./cli-command-generator');
+
+/**
+  CliCommandCompleter is responsible for shell completion for partially
+  typed - and shell suggestion for fully typed cli commands.
+
+  @class CliCommandCompleter
+  @extends CoreObject
+  @constructor
+*/
+
+function CliCommandCompleter() {
+
+  try {
+    this.completion = omelette('ember');
+    this.template = this.loadTemplate('/node_modules/ember-cli/cli-commands.json');
+  } catch(e) {}
+
+}
+
+/**
+  Receives a relative path to the cli command cache file and returns its
+  content. If no cache file exists it will fallback to generate the JSON
+  on demand.
+
+  @private
+  @method loadTemplate
+ */
+
+CliCommandCompleter.prototype.loadTemplate = function(relPathToCache) {
+
+  var pathToCache = process.cwd() + relPathToCache;
```

Comment:
we must load stuff relative to the dirname, not process. As ember-cli can be run deeply nested..